drivers/shtc1/shtc1.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2017 RWTH-Aachen
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_shtc1
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the Sensirion SHTC1 temperature
+ *              and humidity sensor.
+ *
+ * @author      Steffen Robertz <steffen.robertz@rwth-aachen.de>
+ * @author      Josua Arndt <jarndt@ias.rwth-aachen.de>
+ * @}
+ */
+#include <string.h>
+#include <math.h>
+
+#include "log.h"
+#include "assert.h"
+#include "shtc1.h"
+#include "shtc1_regs.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+#include "assert.h"
+
+/*
+ * does a crc check and returns 0 for passed and -1 for failed
+ * 2 bytes will be checked for the checksum, rec_value is pointer
+ * to first value
+ */
+static int8_t _check_crc(uint8_t *rec_values, uint8_t right_crc)
+{
+    uint8_t crc = 0xFF;
+    uint8_t current_byte;
+    uint8_t bit;
+
+    for (current_byte = 0; current_byte < 2; current_byte++) {
+        crc ^= (rec_values[current_byte]);
+        for (bit = 8; bit > 0; bit--) {
+            if (crc & 0x80) {
+                crc = (crc << 1) ^ SHTC1_CRC;
+            }
+            else {
+                crc = (crc << 1);
+            }
+        }
+    }
```

Comment:
No need to implement this on your own.
If you rebase, you can just `#include "checksum/crc8.h"`